shorebird_code_push/example/lib/main.dart

```diff
@@ -40,34 +42,51 @@ class MyHomePage extends StatefulWidget {
 class _MyHomePageState extends State<MyHomePage> {
   int? _currentPatchVersion;
   bool _isCheckingForUpdate = false;
+  bool get _isShorebirdAvailable => _shorebirdCodePush != null;
 
   @override
   void initState() {
     super.initState();
-    // Request the current patch number.
-    _shorebirdCodePush.currentPatchNumber().then((currentPatchVersion) {
-      if (!mounted) return;
-      setState(() {
-        _currentPatchVersion = currentPatchVersion;
+    if (_isShorebirdAvailable) {
+      // Request the current patch number.
+      _shorebirdCodePush!.currentPatchNumber().then((currentPatchVersion) {
```

Comment:
I don't love this personally because you have to always deal with a nullable instance and sprinkle null checks throughout. What are you thoughts on having two implementations of the `ShorebirdCodePush` interface and when you don't have shorebird integrated you get no-ops/errors.

Reply:
> I'm not 100% clear on what the problem we're solving is tbh.

There have been several Discord questions over the past week where users expressed confusion about the behavior of this plugin due to the silent failures.

Reply:
Is that much different than checking whether `shorebirdCodePush` is null? I feel like it just shifts the null check to the return value instead of the `shorebirdCodePush` instance.